client/src/routes/Session/hooks/useSubscribeToSession.ts

```diff
@@ -1,22 +1,43 @@
 import {useCallback} from 'react';
 import firestore from '@react-native-firebase/firestore';
-import {Session, SessionData} from '../../../../../shared/src/types/Session';
+import {
+  SessionState,
+  Session,
+  SessionStateData,
+  SessionData,
+} from '../../../../../shared/src/types/Session';
 import {getData} from '../../../../../shared/src/modelUtils/firestore';
-import {getSession} from '../../../../../shared/src/modelUtils/session';
+import {
+  getSession,
+  getSessionState,
+} from '../../../../../shared/src/modelUtils/session';
 
 const useSubscribeToSession = (sessionId: Session['id']) => {
   return useCallback(
-    (onSnapshot: (session: Session | undefined) => any) => {
-      const doc = firestore().collection('sessions').doc(sessionId);
+    (
+      onSnapshot: (
+        sessionState: SessionState | undefined,
+        session: Session | undefined,
+      ) => any,
+    ) => {
+      const sessionDoc = firestore().collection('sessions').doc(sessionId);
+      const stateDoc = sessionDoc.collection('state').doc(sessionId);
 
-      const unsubscribe = doc.onSnapshot(
+      const sessionPromise = sessionDoc.get();
```

Comment:
We can replace this by an API call too if we want.

Reply:
Guess not, we a complete firestore doc to be able to subscribe right? What comes from an api call is just data

Reply:
But this is the session part that we don't subscribe to. This should def. be fetched with rest imo.

Reply:
Or fetch the session separate from this hook, since it's actually not a subscription

Reply:
I'll try the navigation param approach 👍